src/wallet/test/wallet_tests.cpp

```diff
@@ -919,10 +919,9 @@ BOOST_FIXTURE_TEST_CASE(wallet_sync_tx_invalid_state_test, TestingSetup)
     // Add tx to wallet
     const auto& op_dest = wallet.GetNewDestination(OutputType::BECH32M, "");
     BOOST_ASSERT(op_dest);
-    const CTxDestination& dest = *op_dest;
 
     CMutableTransaction mtx;
-    mtx.vout.push_back({COIN, GetScriptForDestination(dest)});
+    mtx.vout.push_back({COIN, GetScriptForDestination(*op_dest)});
```

Comment:
Thanks for picking up https://github.com/bitcoin/bitcoin/pull/25721#discussion_r938759439 here.